android/intltest/java/com/facebook/hermes/test/assets/intl_getCanonicalNames_2.js

```diff
@@ -0,0 +1,131 @@
+function assertArray(l, r) {
+  assert(compareArray(l, r), r);
+}
+
+// https://github.com/tc39/test262/blob/master/test/intl402/Intl/getCanonicalLocales/preferred-grandfathered.js
+var irregularGrandfathered = [
+  "en-gb-oed",
+  "i-ami",
+  "i-bnn",
+  "i-default",
+  "i-enochian",
+  "i-hak",
+  "i-klingon",
+  "i-lux",
+  "i-mingo",
+  "i-navajo",
+  "i-pwn",
+  "i-tao",
+  "i-tay",
+  "i-tsu",
+  "sgn-be-fr",
+  "sgn-be-nl",
+  "sgn-ch-de",
+];
+
+var regularGrandfatheredNonUTS35 = [
+  "no-bok",
+  "no-nyn",
+  "zh-min",
+  "zh-min-nan",
+];
+
+var regularGrandfatheredUTS35 = {
+  "art-lojban": "jbo",
+  "cel-gaulish": "xtg-x-cel-gaulish",
+  //"zh-guoyu": "zh",
+  "zh-hakka": "hak",
+  "zh-xiang": "hsn",
+};
+
+
+irregularGrandfathered.forEach(function (tag) {
+  assert.sameValue(
+    isCanonicalizedStructurallyValidLanguageTag(tag), false,
+    "Test data \"" + tag + "\" is not a structurally valid language tag."
+  );
+});
+
+regularGrandfatheredNonUTS35.forEach(function (tag) {
+  assert.sameValue(
+    isCanonicalizedStructurallyValidLanguageTag(tag), false,
+    "Test data \"" + tag + "\" is not a structurally valid language tag."
+  );
+});
+Object.getOwnPropertyNames(regularGrandfatheredUTS35).forEach(function (tag) {
+  var canonicalizedTag = regularGrandfatheredUTS35[tag];
+  assert(
+    isCanonicalizedStructurallyValidLanguageTag(canonicalizedTag),
+    "Test data \"" + canonicalizedTag + "\" is a canonicalized and structurally valid language tag."
+  );
+});
+
+Object.getOwnPropertyNames(regularGrandfatheredUTS35).forEach(function (tag) {
+  var canonicalLocales = Intl.getCanonicalLocales(tag);
+  assert.sameValue(canonicalLocales.length, 1);
+  assert.sameValue(canonicalLocales[0], regularGrandfatheredUTS35[tag]);
+});
+
+
+// https://github.com/tc39/test262/blob/master/test/intl402/Intl/getCanonicalLocales/preferred-variant.js
+var canonicalizedTags = {
+  // "ja-latn-hepburn-heploc": "ja-Latn-alalc97-hepburn", // TODO::Hermes icu4j reurns ja-latn-hepburn-heploc.. while icu4c returns ja-Latn-alalc97-hepburn .. Not V8 parity.
+};
```

Comment:
Combining and modifying the upstream test files creates a burden for upgrading test262. I'd rather use a skiplist.

Reply:
Sure.. In the latets iteration, we don't make any changes to the test262 files.